bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/communication/SonnenJsonDataDTO.java

```diff
@@ -138,4 +142,26 @@ public boolean isFlowProductionBattery() {
     public boolean isFlowProductionGrid() {
         return flowProductionGrid;
     }
+
+    /**
+     * @return the em_operatingMode
+     */
+    public String emgetOperationMode() {
+        return emoperatingMode;
+    }
+
+    /**
+     * @return the operatingMode
+     */
+    public boolean isInAutomaticMode() {
+        // 2 for automatic mode
+        if ("2".equals(operatingMode)) {
+            return true;
+        } else if ("1".equals(operatingMode)) {
+            return false;
+        }
```

Comment:
You could serialize operatingMode as an enum. It would make this a bit more meaningfull, readable and maintainable.

Reply:
serialized it different. prefer not use enum